src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/Interop/SNINativeMethodWrapper.cs

```diff
@@ -17,8 +17,15 @@ namespace Microsoft.Data.SqlClient
 {
     internal static class SNINativeMethodWrapper
     {
+        private static readonly ISniNativeMethods NativeMethods = RuntimeInformation.ProcessArchitecture switch
+        {
+            Architecture.Arm64 => new SniNativeMethodsArm64(),
+            Architecture.X64 => new SniNativeMethodsX64(),
+            Architecture.X86 => new SniNativeMethodsX86(),
+            _ => new SniNativeMethodsNotSupported(RuntimeInformation.ProcessArchitecture)
+        };
```

Comment:
With this being the only major variation between netcore and netfx versions of the class, do you think it'd be viable to merge them in this PR?

Reply:
There's a few more differences between... I could send it all out at once, I just didn't want anyone to get overwhelmed

Reply:
Ok, I've rolled the branch forward to take the rest of the SniNativeWrapper merge (no cleanup) changes